k8s/plugin/src/main.rs

```diff
@@ -171,14 +167,15 @@ async fn execute(cli_args: CliArgs) {
                     UpgradeResources::uninstall(&cli_args.namespace).await;
                 }
             },
-            Operations::Upgrade => {
-                UpgradeResources::apply(
-                    cli_args.upgrade_operator_endpoint,
-                    &cli_args.namespace,
-                    cli_args.kube_config_path,
-                    cli_args.timeout,
-                )
-                .await;
+            Operations::Upgrade(resources) => {
+                console_logger::info(user_prompt::UPGRADE_WARNING.to_string());
+                resources
+                    .implement(
```

Comment:
I don't understand the details here, but IMHO apply seemed better since under the hood is also apply.

Reply:
changed